### PR TITLE
chore: Switch to unbounded anchor memory

### PR DESCRIPTION
**Pull request by aterga**
# Motivation

Lift the 4KB restriction per anchor.

# Changes

`Storage.read` now attempts to read anchors from the new, unbounded stable memory, falling back to legacy memory (for now).

# Tests

Added comprehensive migration test with 18 test cases covering device types, metadata preservation, and edge cases.

### CODE REVIEW

File: src/internet_identity/src/storage/anchor.rs [hunk 1]
```diff
@@ -25,12 +25,12 @@ mod tests;
...
-    metadata: Option<HashMap<String, MetadataEntry>>,
-    name: Option<String>,
-    created_at: Option<Timestamp>,
+    pub(crate) anchor_number: AnchorNumber,
+    pub(crate) devices: Vec<Device>,
+    pub(crate) openid_credentials: Vec<OpenIdCredential>,
+    pub(crate) metadata: Option<HashMap<String, MetadataEntry>>,
+    pub(crate) name: Option<String>,
+    pub(crate) created_at: Option<Timestamp>,
 }
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity/src/storage/anchor.rs:35]: This is needed for Rust not to complain about the new tests.

File: src/internet_identity/src/storage.rs [hunk 2]
```diff
@@ -778,21 +765,38 @@ impl<M: Memory + Clone> Storage<M> {
...
         let storable_fixed_anchor = StorableFixedAnchor::from_bytes(Cow::Owned(buf));
-        let storable_anchor = self.stable_anchor_memory.get(&anchor_number);
 
-        if storable_anchor.is_none() {
+        Ok(Anchor::from((anchor_number, storable_fixed_anchor, None)))
+    }
+
+    /// Reads the data of the specified anchor from stable memory.
+    pub fn read(&self, anchor_number: AnchorNumber) -> Result<Anchor, StorageError> {
+        // These values are no longer used for reading, but we keep the check for consistency.
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity/src/storage.rs:774]: Too pedantic, IMHO

**Review by sea-snake — APPROVED** [review]
LGTM
